arbiter-core/src/environment.rs

```diff
@@ -447,13 +471,38 @@ pub(crate) struct Socket {
     pub(crate) event_broadcaster: Arc<Mutex<EventBroadcaster>>,
 }
 
+/// Represents the possible outcomes of an EVM transaction.
+///
+/// This enum is used to encapsulate both successful transaction results and
+/// potential errors.
+/// - `Success`: Indicates that the transaction was executed successfully and
+///   contains the result of the execution. The wrapped `ExecutionResult`
+///   provides detailed information about the transaction's execution, such as
+///   returned values or changes made to the state.
+/// - `Error`: Indicates that the transaction failed due to some error
+///   condition. The wrapped `EnvironmentError` provides specifics about the
+///   error, allowing callers to take appropriate action or relay more
+///   informative error messages.
+#[derive(Debug, Clone)]
+pub(crate) enum TransactionOutcome {
+    /// Represents a successfully executed transaction.
+    ///
+    /// Contains the result of the transaction's execution.
+    Success(ExecutionResult),
+
+    /// Represents a failed transaction due to some error.
+    ///
+    /// Contains information about the error that caused the transaction
+    /// failure.
+    Error(EnvironmentError),
+}
```

Comment:
For whatever reason I still feel like we can get by without this entirely but I'm not worried about it for now.

Reply:
We can change it in the future, if we want to propagate the Environment errors to the client we will need something like this.